net/socket.cpp

```diff
@@ -32,6 +32,11 @@ std::string read_all_impl(auto &socket) {
     return data;
 }
 
+std::size_t read_until_impl(auto &socket, std::string &data, std::string_view delimiter) {
```

Comment:
May be a nicer API to also return a std::string in this one? It looks like the way it's used in the commit after, every call to read_until is followed by clearing the data with erase and reading into the new empty string? We still get the size by calling string::size() when we need it.

Reply:
I agree that this API is not great. The thing is that asio::read_until appends to the buffer (the string in this case) in chunks and returns once the delimiter is found. So the string may contain more data after the delimiter. The return value is not the number of bytes in the string, it is the position of the delimiter in the buffer (delimiter included). So the erase is only to remove the "consumed" data.

I got an idea while writing this on how we could do to make this return a string instead. I'll do a commit and add it to this PR.

Reply:
Nice! Looks good! I should've looked at the asio API before commenting, but glad you found a solution. :P